base/cvd/cuttlefish/host/libs/web/luci_build_api.cpp

```diff
@@ -31,12 +31,11 @@ LuciBuildApi::LuciBuildApi() : http_client_(HttpClient::CurlClient()) {}
 LuciBuildApi::LuciBuildApi(
     std::unique_ptr<HttpClient> http_client,
     std::unique_ptr<HttpClient> inner_http_client,
-    std::unique_ptr<CredentialSource> buildbucket_credential_source,
-    std::unique_ptr<CredentialSource> storage_credential_source)
+    CredentialSources& credential_sources)
     : http_client_(std::move(http_client)),
       inner_http_client_(std::move(inner_http_client)),
-      buildbucket_credential_source_(std::move(buildbucket_credential_source)),
-      storage_credential_source_(std::move(storage_credential_source)) {}
+      buildbucket_credential_source_(std::move(credential_sources[CredentialSourceId::kLuci])),
+      storage_credential_source_(std::move(credential_sources[CredentialSourceId::kLuciStorage])) {}
```

Comment:
Don't move out a map without removing, it leaves the door open for something else to try to access that object later since technically `Contains(credential_sources, kLuci)` returns `true`.

Reply:
I'll clean this up.